tests/local.rs

```diff
@@ -0,0 +1,264 @@
+                        } else if std::path::Path::new("/Applications/Google Chrome.app/Contents/MacOS/Google Chrome").exists() {
+                            // macOS
+                            "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
+                        } else {
+                            // elsewhere, it's just called chromium
+                            "/usr/bin/chromium"
+                        }
+                });
+                caps.insert("goog:chromeOptions".to_string(), opts.clone());
+
+                Client::with_capabilities("http://localhost:9515", caps)
+            },
+            browser => unimplemented!("unsupported browser backend {}", browser),
+        };
+
+        // we'll need the session_id from the thread
+        // NOTE: even if it panics, so can't just return it
+        let session_id = Arc::new(Mutex::new(None));
+
+        // run test in its own thread to catch panics
+        let sid = session_id.clone();
+        let success = match thread::spawn(move || {
+            let mut rt = tokio::runtime::Builder::new().enable_all().basic_scheduler().build().unwrap();
+            let mut c = rt.block_on(c).expect("failed to construct test client");
+            *sid.lock().unwrap() = rt.block_on(c.session_id()).unwrap();
+            let x = rt.block_on($f(c, port));
+            drop(rt);
+            x
+        })
+        .join()
+        {
+            Ok(Ok(_)) => true,
+            Ok(Err(e)) => {
+                eprintln!("test future failed to resolve: {:?}", e);
+                false
+            }
+            Err(e) => {
+                if let Some(e) = e.downcast_ref::<error::CmdError>() {
+                    eprintln!("test future panicked: {:?}", e);
+                } else if let Some(e) = e.downcast_ref::<error::NewSessionError>() {
+                    eprintln!("test future panicked: {:?}", e);
+                } else {
+                    eprintln!("test future panicked; an assertion probably failed");
+                }
+                false
+            }
+        };
+
+        assert!(success);
+    }};
+}
+
+lazy_static::lazy_static! {
+    static ref PORT_COUNTER: std::sync::atomic::AtomicU16 = std::sync::atomic::AtomicU16::new(8000);
+}
+
+
+fn setup_server() -> u16 {
+    let port: u16;
+    loop {
+        let prospective_port = PORT_COUNTER.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        if port_scanner::local_port_available(prospective_port) {
+            port = prospective_port;
+            break
+        }
+    }
+
+    std::thread::spawn(move || {
+
+        let mut rt = tokio::runtime::Builder::new().enable_all().basic_scheduler().build().unwrap();
+        let server = start_server(port);
+        rt.block_on(server);
+    });
+    std::thread::sleep(std::time::Duration::from_secs(1));
+    port
+}
+
+
+
+/// Starts the fileserver
+async fn start_server(port: u16) {
+    let localhost = [0, 0, 0, 0];
```

Comment:
You probably want [this instead](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#associatedconstant.LOCALHOST).